render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
I think thats a pretty interesting idea. We could even make it a requirement of the responder. 

```golang
func routes() http.Handler {
  r := chi.NewRouter()
  r.Use(middleware.Logger)
  r.Use(render.Responder)
  r.Get(....)
  return r
}

func h(w http.ResponseWriter, r *http.Request) {
  // a call to Respond() would just put the response status/body on the r.Context()
  // and leave it up to the Responder to return..
  // in fact, Respond() could detect that Responder was set in the chain, because
  // it could check for a context value, and either respond right there, or defer
  // to the Responder middleware
  render.Respond(w, r, 200, "something")
}
```